src/classes/ADDR_Contact_TEST.cls

```diff
@@ -0,0 +1,1112 @@
+            system.assert(contact.MailingState.contains('Oregon')); 
+            system.assertEquals(false, contact.is_Address_Override__c);
+            system.assertNotEquals(null, contact.Current_Address__c);
+        }
+ 
+        // verify that no new address was created
+        List<Address__c> ListAddr = [select Id, Default_Address__c, MailingStreet__c, Parent_Contact__c from Address__c];
+        system.assertEquals(2, ListAddr.size());
+        for (Address__c addr : ListAddr) {
+            system.assertEquals('direct street  edit', addr.MailingStreet__c);
+        }        
+    }
+
+    /*********************************************************************************************************
+    @description Create an Address with no Contact 
+    verify: runtime error
+    **********************************************************************************************************/            
+    @isTest
+    public static void newAddrNoParent() {        
+        List<Address__c> addrs = UTIL_UnitTestData_TEST.initTestAddr(2);
+        for (Integer i = 0; i < addrs.size(); i++) {
+            system.assertEquals(null, addrs[i].Parent_Contact__c);
+        }
+        
+        try {
+            insert addrs;
+        } catch (Exception ex) {
+            system.assert(ex.getMessage().contains(Label.addrValidParentObjects));
+            return;
+        }
+        system.assert(false); //we shouldn't get here!
+    }
+ 
+    /*********************************************************************************************************
+    @description Create N new duplicate Addresses from Contact  
+    verify:
+        duplicates not created
+    **********************************************************************************************************/            
+    @isTest
+    public static void newDupeAddr() {
+        configSettings();
+        
+        // this creates a default Address for each Contact
+        UTIL_UnitTestData_TEST.ContactsWithAddrs contactsAddrs = UTIL_UnitTestData_TEST.createTestContactsAddrs(2);
+
+        // create additional addresses thru the Contact.
+        for (Integer i = 0; i < contactsAddrs.contacts.size(); i++) {
+            contactsAddrs.contacts[i].MailingStreet = contactsAddrs.addrs[i].MailingStreet__c;
+            contactsAddrs.contacts[i].MailingCity = contactsAddrs.addrs[i].MailingCity__c;
+            contactsAddrs.contacts[i].MailingState = contactsAddrs.addrs[i].MailingState__c;
+            contactsAddrs.contacts[i].MailingPostalCode = contactsAddrs.addrs[i].MailingPostalCode__c;
+            contactsAddrs.contacts[i].MailingCountry = contactsAddrs.addrs[i].MailingCountry__c;
+        }        
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        Test.startTest();
+        update contactsAddrs.contacts;
+        Test.stopTest();
+        
+        // verify that we don't have new Addresses
+        List<Address__c> ListAddr = [select Id from Address__c];
```

Comment:
If the table in this query contains more than 50'000 records a runtime exception will be thrown. I suggest to introduce a `WHERE` clause or cap the max number of records returned adding a `LIMIT` clause to the query.